src/domain/contexts/ChatbotContext.tsx

```diff
@@ -0,0 +1,151 @@
+import React, { createContext, ReactNode, useContext, useEffect, useMemo, useState } from 'react';
+
+import { IChat } from '@/domain/entities/chatbot/Chatbot';
+import { SenderRole } from '@/domain/enums/SenderRole';
+import ChatbotOperations from '@/operations/chatbot/Chatbot';
+import { useMediaQuery } from '@/utils/resolution';
+
+interface ChatbotContextType {
+  chats: IChat[];
+  setChats: React.Dispatch<React.SetStateAction<IChat[]>>;
+  currentChatIndex: number;
+  setCurrentChatIndex: React.Dispatch<React.SetStateAction<number>>;
+  openChat: (chatIndex: number) => void;
+  startNewChat: (newChat?: IChat) => void;
+  chatWithReport: (countryName: string, report: string) => Promise<void>;
```

Comment:
chatWithReport is a little bit weird naming. Can it be more precise? At least I don't understand what it is about